TASVideos/Pages/Profile/Settings.cshtml

```diff
@@ -148,45 +163,86 @@
 	<partial name="_ValidationScriptsPartial" />
 	<script>
 		let avatarBoxElem = document.getElementById('@Html.IdFor(m => m.Avatar)');
-		let previewElem = document.getElementById('avatar-preview');
 
 		document.addEventListener("DOMContentLoaded", validateAvatar);
-		avatarBoxElem.addEventListener('blur', toggleAvatarPreview);
-		document.getElementById('preview-btn').addEventListener('click', toggleAvatarPreview);
-
+		document.addEventListener("DOMContentLoaded", onGravatarToggle);
+		avatarBoxElem.addEventListener('input', generateAvatarPreview);
+		document.getElementById('gravatar-email').addEventListener('input', generateGravatarPreview)
 		let avatarImgElem = document.getElementById('avatar-img');
+		avatarImgElem.onload = validateAvatar;
+
+		Array.from(document.querySelectorAll('[name="UseGravatar"]')).forEach(elem => elem.addEventListener('click', onGravatarToggle));
 
-		function toggleAvatarPreview() {
+		function generateAvatarPreview() {
 			const avatar = document.getElementById('@Html.IdFor(m => m.Avatar)').value;
 			if (avatar) {
-				avatarImgElem.onload = validateAvatar;
 				avatarImgElem.src = avatar;
 			} else {
-				previewElem.classList.add('d-none');
+				avatarImgElem.src = '';
 				preventSave(false);
 			}
 		}
 
 		function validateAvatar() {
+			console.log('validating avatar requirements');
```

Comment:
We probably don't want stray console.log's.

Reply:
Yup, that was an accident, fixed